### DISCUSSION TITLE
Getting "^[[0K"  at end of some lines

### DISCUSSION THREAD

**Discussion by NicholasSterling** [dandavison/delta#1630] [Q&A] [answered]
Hi folks.  I just installed git-delta and told git to use it.  The output looks fine with git diff, but if I just use delta to compare two files (no switches), I get a lot of lines ending in "^[[0K" .  I think all new lines and all modified lines have it.  This is using iTerm2 on Mac OS.  Any idea what the problem is?  I'm guessing it has something to do with color, but it's interesting that git diff in an iTerm2 does not have the problem.  Thanks for any help.
![Screenshot 2024-02-18 at 11 26 58 PM](https://github.com/dandavison/delta/assets/1130633/1a3eba00-5a54-4735-8be3-98716fcb0326)

Edit: I just discovered that the problem does NOT occur with WezTerm.

**Comment by nonZero** [comment]
also happens on arch linux + zsh, delta 0.16.5

**Comment by WayneD** [comment]
That is the escape code for clearing to the end of the line, though I'm not sure why Dan decided to force the `0` in there. The idea is that a background color is currently set as the default, and outputting Esc-`[0K` should color the rest of the line with the background color without having to manually fill it with trailing spaces.  Perhaps your terminal would handle the escape without the `0`? You could try tweaking this line and doing a fresh build:

```
src/ansi/mod.rs:13
pub const ANSI_CSI_CLEAR_TO_EOL: &str = "\x1b[0K";
```

Finally, you should mention what terminal software you're using (instead of the OS) as that is what is interpreting the escape codes. Perhaps a bug report could be filed with them at some point.

**Reply by NicholasSterling** [reply, marked as answer]
Thanks for the explanation, Wayne.  I did mention that I was using iTerm2, but I did not say the version.  I just tried to reproduce it on iTerm2 build 3.4.23, but it appears to be fixed.  I did not know until now because I switched to WezTerm after creating this post.

**Comment by khughitt** [comment]
Similar issue on arch / zsh / delta 0.18.2 / kitty 0.36.2 except the trailing escape code appears as "[0K" instead of "^[[0K".
